### PR TITLE
Bug fix:  `undefined` appearing in reference field

### PR DESCRIPTION
**Pull request by KamBustos**
When loading the page, specifically in the Canonical Reference Locator section, there is a bug where the user sees undefined. For example, Canonical Reference Locator: Matthew:undefined. undefined should not be appearing there. Also, see Issue https://github.com/CenterForDigitalHumanities/glossing-entries/issues/79 in relation to optimizing and cleaning code up.

Example of Bug:
image

Made some changes in the code so that it checks for the primitive type of `undefined` on page load.
I edited solely the file ng.js.
Test the code by loading the page of the new gloss tab. It works correctly if `undefined` does not appear.

### CODE REVIEW

**Comment by AbhilashKotha** [comment]
@KamBustos Great work. The bug is fixed for me. @thehabes can you take a look at this before I merge it? Thanks.

File: js/ng.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ addEventListener('deer-form-rendered', event => {
             prefillThemesArea(annotationData["themes"], event.target)
             prefillText(annotationData["text"], event.target)
             if(event.detail.targetChapter && !event.detail.section) {
-                document.querySelector('[deer-key="canonicalReference"]').value = `Matthew ${event.detail.targetChapter.value}:${event.detail.targetVerse.value}`
+                document.querySelector('[deer-key="canonicalReference"]').value = `Matthew ${event.detail.targetChapter.value || ''}${event.detail.targetVerse.value ? `:${event.detail.targetVerse.value}` : ''}`
```
Comment (thehabes) [inline js/ng.js:80]: From testing, I no longer see any `undefined` appearances!

Looks like the right place and the right logic.  Do both the checks of `targetVerse` and `targetChapter` in the same syntax style. 

```javascript

 document.querySelector('[deer-key="canonicalReference"]').value = `Matthew ${event.detail.targetChapter.value ?  event.detail.targetChapter.value : ''}${event.detail.targetVerse.value ? `:${event.detail.targetVerse.value}` : ''}`
 
```

**Review by thehabes — CHANGES_REQUESTED** [review]

File: js/ng.js (hunk 1 above)
Comment (thehabes) [reply js/ng.js:80]: Oh nevermind!  That extra `:` you add in is why you didn't do them both like ` || ''`

**Review by thehabes — APPROVED** [review]

**Review by AbhilashKotha — APPROVED** [review]
lgtm

**Pull request merged**